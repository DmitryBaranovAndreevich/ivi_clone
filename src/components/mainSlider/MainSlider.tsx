import styles from './mainSlider.module.scss';
import 'keen-slider/keen-slider.min.css';
import { useKeenSlider } from 'keen-slider/react';
import SliderButton from '../UI/sliderButton/SliderButton';
import { CSSProperties, FC, useEffect, useState } from 'react';
import { mediaQuery } from '../../service/constans';
import RedButton from '../UI/redButton/RedButton';
import { IFilm } from '../../type/TFilm';

interface IMainSlider {
  items: IFilm[];
}

const MainSlider: FC<IMainSlider> = ({ items }) => {
  const [dark, setDark] = useState(true);
  const [sliderRef, instanceRef] = useKeenSlider({
    initial: 1,
    mode: 'free-snap',
    slides: {
      origin: 'center',
      perView: 1.2,
      spacing: 60,
    },
    defaultAnimation: {
      duration: 1000,
    },
    animationStarted: () => setDark(false),
    dragEnded: () => setTimeout(() => setDark(true), 280),
    breakpoints: {
      [mediaQuery.$laptop]: {
        slides: {
          origin: 'center',
          perView: 1.2,
          spacing: 42,
        },
      },
      [mediaQuery.$phones]: {
        slides: {
          origin: 'center',
          perView: 1.2,
          spacing: 32,
        },
      },
      [mediaQuery.$tablets]: {
        slides: {
          origin: 'center',
          perView: 1.2,
          spacing: 22,
        },
      },
    },
  });

  const callback = () => {
    instanceRef.current?.update();
  };

  useEffect(() => {
    window.addEventListener('resize', callback);
    return () => window.removeEventListener('resize', callback);
  }, []);

  return (
    <>
      <div
        ref={sliderRef}
        className={`${styles.container} keen-slider ${dark && styles.container_darkBackground}`}
      >
        <div className={styles.buttonContainer}>
          <SliderButton
            direction={'right'}
            addClass={styles.rightButton}
            onClick={() => {
              instanceRef.current?.next();
            }}
          />
          <SliderButton
            direction={'left'}
            addClass={styles.leftButton}
            onClick={() => {
              instanceRef.current?.prev();
            }}
          />
        </div>
        {items.map((movie) => (
          <div
            className={`${styles.slide} keen-slider__slide`}
            style={{ '--image': `url(${movie.poster})` } as CSSProperties}
            key={movie.id}
          >
            <p className={styles.moviesName}>{movie.name}</p>
            <p className={styles.description}>{movie.description}</p>
            <div>
              <RedButton addingClass={styles.button} text={'Смотреть по подписке'} />
            </div>
          </div>
        ))}
      </div>
    </>
  );
};

export default MainSlider;
