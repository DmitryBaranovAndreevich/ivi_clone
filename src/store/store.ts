import reviewReducer from './reducers/ReviewSlice';
import moviesSortReduser from './reducers/MoviesSort';
import { appApi } from './api/appApi';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import userLoginReduser from './reducers/UserLoginSlice';
import appReducer from './reducers/App';
import moviesFilterReduser from './reducers/MoviesFilter';
import { filmApi } from './api/filmApi';
import { personApi } from './api/personApi';
import { reviewApi } from './api/reviewApi';

const rootReducer = combineReducers({
  appReducer,
  userLoginReduser,
  moviesSortReduser,
  reviewReducer,
  moviesFilterReduser,
  [appApi.reducerPath]: appApi.reducer,
  [filmApi.reducerPath]: filmApi.reducer,
  [personApi.reducerPath]: personApi.reducer,
  [reviewApi.reducerPath]: reviewApi.reducer,
});

export const setupStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat([appApi.middleware, filmApi.middleware, personApi.middleware]),
  });
};

export type TRootState = ReturnType<typeof rootReducer>;
export type TAppStore = ReturnType<typeof setupStore>;
export type TAppDispatch = TAppStore['dispatch'];
