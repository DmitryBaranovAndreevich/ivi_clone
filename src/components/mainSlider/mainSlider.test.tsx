import { BrowserRouter } from "react-router-dom";
import { store } from '../../store/mainStore';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import MainSlider from './MainSlider';
import renderer from 'react-test-renderer';
describe('Тестим компонент слайдера большого размера', () => {

  it('Тест слайдера', async () => {
    const tree = renderer.create(<BrowserRouter><Provider store={store}><MainSlider items={[]} /></Provider></BrowserRouter>);
    expect(tree).toMatchSnapshot();
  })
})