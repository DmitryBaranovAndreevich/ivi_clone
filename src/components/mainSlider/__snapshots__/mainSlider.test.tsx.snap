// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестим компонент слайдера большого размера Тест слайдера 1`] = `
<div
  className="undefined keen-slider undefined"
>
  <div>
    <div
      className="undefined undefined"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="68"
        viewBox="0 0 10 18"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 17L8 9L1 0.999999"
          opacity="0.75"
          stroke="#C7C3C3"
          strokeWidth="2"
        />
      </svg>
    </div>
    <div
      className="undefined undefined"
      onClick={[Function]}
    >
      <svg
        fill="none"
        height="58"
        viewBox="0 0 10 18"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 1L2 9L9 17"
          stroke="#C7C3C3"
          strokeWidth="2"
        />
      </svg>
    </div>
  </div>
</div>
`;
